runtime/src/main/java/org/corfudb/protocols/CorfuProtocolCommon.java

```diff
@@ -103,10 +110,18 @@ public static LayoutMsg getLayoutMsg(Layout layout) {
     /**
      * Returns a Layout object from its Protobuf representation.
      *
-     * @param msg   the Protobuf Layout message
+     * @param msg   the desired Protobuf Layout message
+     * @return      an equivalent, potentially null, Java Layout object
+     * @throws      SerializerException if unable to deserialize the JSON payload
      */
     public static Layout getLayout(LayoutMsg msg) {
-        return Layout.fromJSONString(msg.getLayoutJson());
+        try {
+            return Layout.fromJSONString(msg.getLayoutJson());
+        } catch (NullPointerException npe) {
```

Comment:
The method looks weird. In what situation, you'd have NPE?

Reply:
There are some cases in the code base that send a null layout. I think @xcchang and I came across this in `LayoutServerTest`. In the `LayoutServer`, for example the `LAYOUT_PREPARE` rpc, the method `getProposedLayout` can return null. This gets converted into an empty json string I believe.

Reply:
We found that more than half of the unit tests within LayoutSeverTest will produce a `null` value for `proposedLayout` during the RPCs are handling on the server side (`prepareRejectsLowerRanks` and `handleMessageLayoutPrepare` for example). In the LayoutSever there are multiple cases that `proposedLayout == null ?` is checked, so we also added a checking here to keep the current logic.

Reply:
let's check for null instead of catching NPE

Reply:
I thought they are equivalent here, since the `Layout.fromJSONString` has a `requireNonNull` constraint for the given argument, and that's where the NPE gets thrown.